packages/ckeditor5-ui/src/editorui/editorui.ts

```diff
@@ -156,6 +157,9 @@ export default abstract class EditorUI extends /* #__PURE__ */ ObservableMixin()
 		this.poweredBy = new PoweredBy( editor );
 		this.ariaLiveAnnouncer = new AriaLiveAnnouncer( editor );
 
+		this.decorate( 'setEditableElement' );
+		this.decorate( 'removeEditableElement' );
+
```

Comment:
I'd strongly discourage exposing additional decorated methods. It's a public API and decoration comes with some cost. I'd suggest using an approach similar to:

https://github.com/ckeditor/ckeditor5/blob/2f8b8f1bebc2119000ef52430077b2141fa81c72/packages/ckeditor5-editor-multi-root/src/multirooteditor.ts#L119

Reply:
@niegowski I added this because in some of our integrations we decoupled creating roots and assigning them roots. Like [here](https://github.com/ckeditor/ckeditor5-react/blob/master/src/useMultiRootEditor.tsx#L615), where creation of root and assigning editable might be delayed. Will the solution above work in that case?

Reply:
@niegowski I added this because in some of our integrations we decoupled creating roots and assigning them editables. Like [here](https://github.com/ckeditor/ckeditor5-react/blob/master/src/useMultiRootEditor.tsx#L615), where creation of root and assigning editable might be delayed. There is also a case where editables are changed without changing roots. I checked this approach using these calls:

```ts
		roots.on<CollectionAddEvent<ViewRootEditableElement>>( 'add', ( evt, viewRootEditable ) => {
			const editableElement = ui.getEditableElement( viewRootEditable.name );

			if ( editableElement ) {
				focusTracker.add( editableElement );
			}
		} );

		roots.on<CollectionRemoveEvent<ViewRootEditableElement>>( 'remove', ( evt, viewRootEditable ) => {
			const editableElement = ui.getEditableElement( viewRootEditable.name );

			if ( editableElement ) {
				focusTracker.remove( editableElement );
			}
		} );
```

and this test that simulates React integration behaviour is [failing](https://github.com/ckeditor/ckeditor5/pull/16767/commits/3f4ad6a190a749c985bc328ce925b8d39b8e7e57#diff-683a0ba95ad6a14b7eaebc593f0e03bc9cc498237b0c53f7b521a76a2cc0d038R947).

Reply:
I found different API that seems to be working exactly like I want. I removed decoration of methods. 

```tsx
		editing.view.addObserver( class TrackEditableElements extends Observer {
			/**
			 * @inheritDoc
			 */
			public observe( domElement: HTMLElement ) {
				focusTracker.add( domElement );
			}

			/**
			 * @inheritDoc
			 */
			public stopObserving( domElement: HTMLElement ) {
				focusTracker.remove( domElement );
			}
		} );
```